lib/sqlalchemy/orm/_orm_constructors.py

```diff
@@ -734,6 +735,54 @@ class of a particular set of mapped classes, to which the rule
     )
 
 
+class RelaionshipConstructorType(Protocol):
```

Comment:
I would remove this, automap is not that central of a tool and the function this is used for is just one small part of a little-used API, we dont need to maintain a large type like this